autoload/neomake/makers/ft/clojure.vim

```diff
@@ -0,0 +1,27 @@
+function! neomake#makers#ft#clojure#EnabledMakers() abort
+    let makers = []
+    if executable('clj-kondo')
+        call add(makers, 'clj_kondo')
+    endif
+    return makers
+endfunction
+
+function! neomake#makers#ft#clojure#clj_kondo() abort
+    let maker = {
+        \ 'exe': 'clj-kondo',
+        \ 'args': ['--lint', '%t'],
+        \ 'append_file': 0,
+        \ 'errorformat': '%f:%l:%c: Parse %t%*[^:]: %m,%f:%l:%c: %t%*[^:]: %m',
+        \ 'output_stream': 'stdout',
+        \ }
+
+    function! maker.supports_stdin(_jobinfo) abort
+        let self.args += ['--filename', '%:p']
+        let self.tempfile_name = '-'
+        return 1
+    endfunction
```

Comment:
`supports_stdin` is not used with `append_file=0`.
(not sure if this is good, and should get documented at least probably).

I've pushed some suggested change, and also added tests - feel free to extend/revisit this/them yourself, of course.. :)
I suggest using `'args': ['--lint', '-', ]`

Reply:
> supports_stdin is not used with append_file=0.

This makes a lot of sense, spent a good bit of time banging my head on this :).  Docs would have been helpful!

>  I suggest using 'args': ['--lint', '-', ]

I was worried about this, because the user might disable stdin, or the file might be linted directly as it's been saved?  I was trying to support use with temp files simultaneously with stdin as I thought that was a thing.  If not, then I guess we could forgo the function and just do `'args': ['--lint', '-', '--filename', '%']`?

Reply:
doc update: https://github.com/neomake/neomake/pull/2519

As for `args': ['--lint', '-', ]` this PR uses a different approach now, which should be fine(r).